gap/attr.gi

```diff
@@ -1653,6 +1653,75 @@ function(D)
     return C;
 end);
 
+# Compute for a given rotation system the facial walks
+InstallMethod(FacialWalks, "for a digraph and a list",
+[IsDigraph, IsList],
+function(D, rotationSystem)
+
+    local FacialWalk, facialWalks, remEdges, cycle;
+
+    if not IsEulerianDigraph(D) then
+        Error("The given digraph is not Eulerian.");
+    fi;
+
+    if not IsDenseList(rotationSystem) or Length(rotationSystem)
+      <> DigraphNrVertices(D) then
+        Error("The given rotation system does not fit to the given digraph.");
+    fi;
+
+    if Difference(DuplicateFreeList(Flat(rotationSystem)), DigraphVertices(D))
+      <> [] then
+        Error("The given rotation system does not fit to the given digraph.");
```

Comment:
Probably better to say something like `the 2nd argument (dense list <rotationSystem>)
is not a rotation system for the 1st argument (digraph <D>), expected the union to be ", DigraphVertices(D), " but found ", Union(rotationSystem)`